synapse/util/caches/lrucache.py

```diff
@@ -119,11 +127,36 @@ async def _expire_old_entries(clock: Clock, expiry_seconds: int) -> None:
 
     logger.debug("Searching for stale caches")
 
+    evicting_due_to_memory = False
+
+    # determine if we're evicting due to memory
+    jemalloc_interface = get_jemalloc_stats()
+    if jemalloc_interface and autotune_config:
+        try:
+            jemalloc_interface.refresh_stats()
+            mem_usage = jemalloc_interface.get_stat("allocated")
+            if mem_usage > max_cache_memory_usage:
+                logger.info("Begin memory-based cache eviction.")
+                evicting_due_to_memory = True
+        except Exception:
+            logger.warning(
+                "Unable to read allocated memory, skipping memory-based cache eviction."
+            )
+
```

Comment:
```suggestion
            mem_usage = jemalloc_interface.get_stat("allocated")
        except Exception:
            logger.warning(
                "Unable to read allocated memory, skipping memory-based cache eviction."
            )
            
        if mem_usage > max_cache_memory_usage:
            logger.info("Begin memory-based cache eviction.")
            evicting_due_to_memory = True
```

When we're doing a broad `try/catch` its worth keeping the smallest amount of logic within the try block, otherwise its easy to accidentally catch and discard programming errors that we didn't mean to catch

Reply:
Just a question about this, and perhaps I don't understand exceptions as well as I should, but since we catch the exception, wouldn't the code keep executing, which means that if `jemalloc_interface.get_stat()` throws an exception then `mem_usage` won't get set, which means that the `if` clause on line 148 will error?